src/wallet/wallet_rpc_server.cpp

```diff
@@ -1099,6 +1032,73 @@ namespace tools
     {
       return false;
     }
+    
+    if (req.stake_transfer) {
+      crypto::signature supernode_signature;
+      if (!epee::string_tools::hex_to_pod(req.supernode_signature, supernode_signature))
+      {
+        er.code = WALLET_RPC_ERROR_CODE_WRONG_SIGNATURE;
+        er.message = "failed to parse supernode signature";
+        return false;
+      }
+
+      crypto::public_key W;
+      if (!epee::string_tools::hex_to_pod(req.supernode_public_id, W) || !check_key(W))
+      {
+        er.code = WALLET_RPC_ERROR_CODE_WRONG_SUPERNODE_KEY;
+        er.message = "invalid supernode public identifier";
+        return false;
+      }
+      const cryptonote::account_public_address& supernode_public_address = dsts.front().addr;
+      std::string supernode_public_address_str = cryptonote::get_account_address_as_str(m_wallet->testnet(), supernode_public_address);
+      std::string data = supernode_public_address_str + ":" + req.supernode_public_id;
+      crypto::hash hash;
+      crypto::cn_fast_hash(data.data(), data.size(), hash);
+
+      if (!crypto::check_signature(hash, W, supernode_signature))
+      {
+        er.code = WALLET_RPC_ERROR_CODE_WRONG_SIGNATURE;
+        er.message = "invalid supernode signature";
+        return false;
+      }
+
+      auto current_height = m_wallet->get_blockchain_current_height();
+      if (req.unlock_time < current_height + config::graft::STAKE_MIN_UNLOCK_TIME_FOR_WALLET)
+      {
+        er.code = WALLET_RPC_ERROR_CODE_GENERIC_TRANSFER_ERROR;
+        er.message = "unlock_time is too low";
+        return false;
+      }
+
+      if (req.unlock_time > current_height + config::graft::STAKE_MAX_UNLOCK_TIME)
+      {
+        er.code = WALLET_RPC_ERROR_CODE_GENERIC_TRANSFER_ERROR;
+        er.message = "unlock_time is too high";
+        return false;
+      }
+
+      if (dsts.size() != 1)
```

Comment:
`transfer_split` expects more than one destinations on input, but in case "stake_transfer" there should be one destination. It looks confusing, please move this functionality to `transfer`

Reply:
Also, please squash your commits related to `stake_transfer` into one commit